src/buf_writer.c

```diff
@@ -134,3 +144,25 @@ buf_writer_terminate(tsdn_t *tsdn, buf_writer_t *buf_writer) {
 		buf_writer_free_internal_buf(tsdn, buf_writer->buf);
 	}
 }
+
+void
+buf_writer_set_allow_auto_flush(buf_writer_t *buf_writer,
+    bool allow_auto_flush) {
+	buf_writer->allow_auto_flush = allow_auto_flush;
```

Comment:
When switching auto_flush to false, also invoke _commit?

Reply:
Makes sense. That actually fixes a bug in my previous version: if a single item overflowed, then the item wouldn't be dumped at all even after the `buf_writer_set_allow_auto_flush(&buf_writer, true)` call, because the `overflowed` flag was still `true`.